infra/terraform/modules/data_access/users/users.py

```diff
@@ -0,0 +1,75 @@
+import json
+import os
+
+import boto3
+
+
+'''
+Environment variables:
+ - STAGE, e.g. "dev", "alpha", etc...
+ - SAML_PROVIDER_ARN, SAML provider used to login
+'''
+
+
+def create_user_role(event, context):
+    """
+    Creates the role for the given user
+
+    event = {"username": "alice"}
+    """
+    trust_relationship = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "Federated": os.environ["SAML_PROVIDER_ARN"],
+                },
+                "Action": "sts:AssumeRoleWithSAML",
+                "Condition": {
+                    "StringEquals": {
+                        "SAML:aud": "https://signin.aws.amazon.com/saml"
+                    }
+                }
+            }
+        ]
+    }
```

Comment:
I think this is a good example of "good duplication" in tests which I was trying to explain earlier. 

The nice thing of having it duplicated in the tests is that we don't rely on a function in the tested code to be correct, e.g. if we would break the `trust_relationship()` function, the test would fail (as it should).